### ISSUE TITLE
Error semantics in Log Data Model don't match general semantics for Exceptions

### ISSUE THREAD

**Issue by GrahamLea** [open-telemetry/opentelemetry-specification#1868]
**What are you trying to achieve?**

Figure out what the correct/proposed OTel attribute names in a structured log should be for information about an exception.

**Additional context.**

The [Log Data Model spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md) doesn't mention exceptions at all. However, [the example mapping for Elastic Common Schema](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#elastic-common-schema) recommends a number of attributes beginning with `error.`

There is already [another spec documenting semantic conventions for Exceptions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md), and they are different. 
It seems like these should be used in the Elastic mapping, as well as mentioned elsewhere in the Log Data Model spec.

**Comment by Oberon00** [comment]
Exceptions are not necessarily errors, even though they are often conflated. You could have an exception that is not an error (cf. Python StopIteration, GeneratorExit) or an error that is not an exception (integer return code, ...).

**Comment by GrahamLea** [comment]
> Exceptions are not necessarily errors, even though they are often conflated.

That's true. I'm not sure the distinction is beneficial in the domain of logging. How often are people likely to log an exception, unless it caused/indicates an error?
Anyway, that's your prerogative.

>  I doubt the exception.stacktrace ... is the same as error.stacktrace

Actually, it's `error.stack_trace`. 😉
It's appears that those attributes are a copy/paste straight from the Elastic Schema. That's fine if that's the right thing to do in that mapping.

The main problem I think is that the only mention of either errors or exceptions (outside of severity levels) is in the mappings - mostly in the Elastic section, with a brief mention under CloudTrail. And where they are mentioned in the Elastic mapping, the convention is based off the Elastic schema. There's no hint for someone reading the Log Data Model page to know that that OTel has semantic conventions for exceptions (and that they're different to the Elastic mapping) without going looking around for them.

**Comment by Oberon00** [comment]
> I'm not sure the distinction is beneficial in the domain of logging. How often are people likely to log an exception, unless it caused/indicates an error?

You were referencing the trace semantic conventions for exceptions, and there an auto-instrumentation can usually not know if an exception was an error or not. I encourage you to also look a the PR/issue history of the exception semantic conventions and error handling. It was probably the most-debated semantic convention in OpenTelemetry, there are lot of different opinions in that area.

**Comment by Oberon00** [comment]
Sorry that last comment might have sounded like "I know better than you!". I was just referencing the trace semantic conventions discussions because they contained quite some interesting thoughts about error vs exception and who should mark something as error (app/instrumentation/backend). But the trace semantic conventions side of the issue is the only thing I know about, I'm rather clueless about the logging spec or ECS conventions.

So I think the initial issue

> Figure out what the correct/proposed OTel attribute names in a structured log should be for information about an exception.

is worthwhile to solve, so I think you should keep the issue open. Unless there was another reason you closed it.

**Comment by GrahamLea** [comment]
Thanks for the apology. This is my first contribution to the project and it felt like I landed straight into an argument, which I don't have the energy for right now. Sorry if my own comments appeared combative.

FWIW, the Logs spec references the trace semantics and says it should also follow them.

**Comment by tigrannajaryan** [comment]
@zenmoto @jkwatson this is related to the topic we discussed in Slack.

I don't immediately see why we cannot use https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md to represent exceptions in Log Records.

**Comment by zenmoto** [comment]
I think we had discussed this at one point, and I don't see any problem with it. We probably just need a pointer somewhere to find this from the logs documentation so it's discoverable.

**Comment by tigrannajaryan** [comment]
@GrahamLea the mapping for Elastic was provided by Elastic folks and if I remember correctly it was written before we had conventions for exceptions (I may be wrong).

If logs decide to adopt the existing exception representation convention for traces then the mapping may be adjusted to reflect that. However, this is a non-trivial matter. We are essentially talking about mapping one telemetry schema (Elastic) to another (OpenTelemetry). It was one of the possible uses cases I had in my mind when designing the concept of schemas https://github.com/open-telemetry/oteps/blob/main/text/0152-telemetry-schemas.md so it is a possible future area of work.

For now the elasticsearch exporter in the Collector does not attempt to do anything like that. It keeps the attributes as is: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0d78df6b0455fd685cf6d9dcb654986e7de7ac86/exporter/elasticsearchexporter/model.go#L50

**Comment by GrahamLea** [comment]
Thanks for explaining the history there, @tigrannajaryan.

For me, the correctness or otherwise of the Elastic mapping is a secondary issue. The main unmet requirement is still:

> Figure out what the correct/proposed OTel attribute names in a structured log should be for information about an exception.

As @zenmoto suggested, if the Exceptions section of the trace Attributes semantics is acceptable, I think the logs spec having a paragraph that references that and uses the words "errors" and "exceptions" would suffice. A sub-section of the Attributes field section would probably be a suitable place.

**Comment by GrahamLea** [comment]
I've made the PR above that implements the suggestion if everyone's happy with the approach.

**Comment by kumoroku** [comment]
FWIW, catching up on the discussion just now, I agree that if there are Semantic Conventions for something in OTel, we should incorporate them for Logs. it appears as if we are all in agreement here, which is great.

@Oberon00 said that those very Semantic Conventions have been hotly debated (I did not follow this myself) and right now are marked as Experimental (although this appears to be true for pretty much all the Semantic Conventions I looked at). not sure where the discussion stands today, but as we incorporate those semantics, we need to make sure we keep an eye on changes as the status moves to Stable.
